ci(workflows): update long and network tests to Python 3.13

JigyasuRajput opened the pull request:
### **ci: update long and network tests to use Python 3.13**  
Fixes #4775  
#### **Changes:**  
- Updated `.github/workflows/testing.yml` to replace Python 3.10 with Python 3.13 in **9 places**.  
- This aligns with issue #4775 to improve memory usage and prevent CI failures due to OSV data size increases.  

#### **Why this change?**  
- Python 3.13 has memory optimizations that help mitigate out-of-memory issues in CI.  

#### **Testing & Verification:**  
- CI should run successfully with the updated version.

terriko commented:
Thank you! 

I've approved the tests to run and we'll see if there's any surprise problems.

terriko commented:
ooookay.  Apparently the branch protection rule requires us to not change the test name, so changing it from 3.10 to 3.13 is leaving me with a required test that can never be completed on this PR.  Ugh.  Unfortunately, there's a process for me to get admin privileges and fix the rule, so it'll probably take a bit to get that resolved.  I'll go get that started, though.

JigyasuRajput commented:
> ooookay. Apparently the branch protection rule requires us to not change the test name, so changing it from 3.10 to 3.13 is leaving me with a required test that can never be completed on this PR. Ugh. Unfortunately, there's a process for me to get admin privileges and fix the rule, so it'll probably take a bit to get that resolved. I'll go get that started, though.

Ohh okay, do let me know if any help/update is required!

terriko reviewed (changes requested):
While I'm working on that.  Can you add 3.10 into the short tests and remove 3.13?  Right now this would turn off 3.10 testing entirely, which isn't ideal.

JigyasuRajput commented:
> While I'm working on that. Can you add 3.10 into the short tests and remove 3.13? Right now this would turn off 3.10 testing entirely, which isn't ideal.

I've updated the short tests to include 3.10 and removed 3.13. Let me know if any further tweaks are needed!

terriko commented:
Looks good, thank you!  I've set the tests to run and I'll be back once they're passing, but I think we're good to go unless 3.10 does something weird in the short tests.

terriko reviewed (approved):
Good to merge now.  Thank you!
